File: jdroid-java-core/src/main/java/com/jdroid/java/utils/Sanitizer.java
package com.jdroid.java.utils;

import java.text.Normalizer;

public class Sanitizer {
	
	public static String plainString(String text) {
		String plain = null;
		if (text != null) {
			String nfdNormalizedString = Normalizer.normalize(text, Normalizer.Form.NFD);
			plain = nfdNormalizedString.replaceAll("[^a-zA-Z0-9\\s]", "");
			plain = plain.replaceAll("\\s{2,}", " ").trim();
		}
		return plain;
	}

	public static String plainStringWithoutExtraSpaces(String text) {
		String plain = plainString(text);
		if (text != null) {
			plain = plain.replaceAll("\\s{2,}", " ").trim();
		}
		return plain;
	}
	
	public static String plainStringWithoutNumbers(String text) {
		String withoutNumbers = removeNumbers(text);
		return plainString(withoutNumbers);
	}
	
	public static String justNumbers(String text) {
		return text != null ? text.replaceAll("[^\\d]", "") : null;
	}
	
	public static String removeNumbers(String text) {
		return text != null ? text.replaceAll("\\d", "") : null;
	}

	public static String removeDashes(String text) {
		return text != null ? text.replaceAll(StringUtils.DASH, "") : null;
	}
}

File: jdroid-java-core/src/main/java/com/jdroid/java/utils/Sanitizer.kt
package com.jdroid.java.utils

import java.text.Normalizer

object Sanitizer {

    fun plainString(text: String?): String? {
        var plain: String? = null
        if (text != null) {
            val nfdNormalizedString = Normalizer.normalize(text, Normalizer.Form.NFD)
            plain = nfdNormalizedString.replace("[^a-zA-Z0-9\\s]".toRegex(), "")
            plain = plain.replace("\\s{2,}".toRegex(), " ").trim { it <= ' ' }
        }
        return plain
    }

    fun plainStringWithoutExtraSpaces(text: String?): String? {
        var plain = plainString(text)
        if (text != null) {
            plain = plain!!.replace("\\s{2,}".toRegex(), " ").trim { it <= ' ' }
        }
        return plain
    }

    fun plainStringWithoutNumbers(text: String): String? {
        val withoutNumbers = removeNumbers(text)
        return plainString(withoutNumbers)
    }

    fun justNumbers(text: String?): String? {
        return text?.replace("[^\\d]".toRegex(), "")
    }

    fun removeNumbers(text: String?): String? {
        return text?.replace("\\d".toRegex(), "")
    }

    fun removeDashes(text: String?): String? {
        return text?.replace(StringUtils.DASH.toRegex(), "")
    }
}
